File: insurance_backend/src/main/java/com/maghrebia/data_extract/DTO/CreateContactDto.java
package com.maghrebia.data_extract.DTO;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class CreateContactDto {

    private String assure;
    private String societe;
    private String telephone;
    private String email;
    private String msh;
    private String motDePasse;
    private String cin;
    private String carteSejour;
    private String passeport;
    private String matriculeFiscale;

    private List<CreateProductionDTO> contracts;
    private List<CreateBanqueDTO> transactions;
}

File: insurance_frontend/src/app/models/create-contact.dto.ts
import { CreateBanqueDTO } from "./create-banque.dto";
import { CreateProductionDTO } from "./create-production.dto";

export interface CreateContactDTO {
  assure: string;
  societe: string;
  telephone: string;
  email: string;
  msh: string;
  motDePasse: string;
  cin: string;
  carteSejour: string;
  passeport: string;
  matriculeFiscale: string;
  contracts: CreateProductionDTO[];
  transactions: CreateBanqueDTO[];
}
